src/main/java/org/linksharing/server/db/user/User.java

```diff
@@ -35,4 +33,14 @@ public class User {
     @Convert(converter = LinksJsonConverter.class)
     private Map<String, Link> links;
 
+    @PrePersist
+    public void setTimeAndLinks() {
+        createdOn = new Timestamp(System.currentTimeMillis());
+        links = new HashMap<>();
+    }
+
+    public String encryptPassword(String password) {
+        setPassword("Encrypted" + password);  //special encryption will be added later
```

Comment:
To hash a password you will need an instance of a `org.springframework.security.crypto.password.PasswordEncoder` (this should be a spring bean). Are you going to add this as a field to the `User` or to pass as a parameter? Both of these options don't look good. Maybe you can think of something else...